### PR TITLE
LIBTD-1068: Log to Graylog server

### PR DESCRIPTION
**Pull request by tingtingjh**

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -72,20 +65,49 @@
...
+        graylog_network_locality, { :protocol => graylog_protocol,
+                                    :facility => graylog_facility })
+    config.log_level = graylog_verbosity
+    config.colorize_logging = false
+    # Log controller params, too
+    config.lograge.custom_options = lambda do |event|
+      params = event.payload[:params].reject { |k| %w(controller action).include?(k) }
+      { "params" => params }
+    end
+  else
```
Comment (pmather) [inline config/environments/production.rb]: Consider using the approach taken in VTUL/data-repo for when Graylog is not enabled: it creates a standard logger that writes logs to that standard log/production.log and rotates them daily (e.g., `config.logger = Logger.new(Rails.root.join('log', "#{Rails.env}.log"), "daily")`). This avoids having to have an external utility like `logrotate` do rotation/management and the attendant application restarts that brings.

Comment (tingtingjh) [reply config/environments/production.rb]: Somehow gelf and lograge gems are already in the Gemfile as well as the current Gemfile.lock

Comment (tingtingjh) [reply config/environments/production.rb]: Changed per your advice.

Comment (tingtingjh) [reply config/environments/production.rb]: Changed per your advice @pmather

File: config/environments/production.rb [hunk 2]
```diff
@@ -72,20 +65,34 @@
...
-  # Use a different logger for distributed setups.
-  # require 'syslog/logger'
-  # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
-
-  if ENV["RAILS_LOG_TO_STDOUT"].present?
-    logger           = ActiveSupport::Logger.new(STDOUT)
-    logger.formatter = config.log_formatter
-    config.logger    = ActiveSupport::TaggedLogging.new(logger)
+  # Logging
+  if Rails.application.secrets['graylog']['enabled'] == true
```
Comment (pmather) [inline config/environments/production.rb]: In my testing, `Rails.application.secrets['graylog']['enabled']` no longer correctly accesses the graylog `enabled` key from `config/secrets.yml` (at least in Rails 5).  The aforementioned returns `nil`.  It appears `Rails.application.secrets[:graylog][:enabled]` is needed to access the actual value.

It's likely the `Rails.application.secrets` code for the other accesses need to be changed, too. If the second-level keys aren't looked up as symbols then the return value is `nil`.

**Comment by tingtingjh** [comment]
@pmather Changed to symbols. Also changed devise setting for consistency.

**Pull request merged**